components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java

```diff
@@ -783,7 +783,9 @@ private APIDTO getAPIByAPIId(String apiId, String xWSO2Tenant) {
             }
 
             API api = apiConsumer.getAPIbyUUID(apiId, requestedTenantDomain);
-            return APIMappingUtil.fromAPItoDTO(api,requestedTenantDomain);
+            if (APIConstants.PUBLISHED.equals(api.getStatus()) || APIConstants.PROTOTYPED.equals(api.getStatus())) {
+                return APIMappingUtil.fromAPItoDTO(api, requestedTenantDomain);
+            }
```

Comment:
We should have an else statement with 403 Forbidden response.

Reply:
Fixed in a1574fcf777f4bc09bad750b316b07e248eff613